features.md

```diff
@@ -0,0 +1,60 @@
+# Features Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+A feature is the representation of a business goal used in the [Authorization endpoint](authorizations.md) to declare what an user can do on a specific object.
+
+## Main Endpoint
+**/api/authz/features**
+
+List all the available features in the system. Access is restricted to system administrators.
+
+Parameters:
+* page, size [see pagination](README.md#Pagination)
+
+Return codes:
+* 200 OK - if the operation succeed
+* 401 Forbidden - if you are not authenticated
+* 403 Unauthorized - if you are not logged in with sufficient permissions. Only system administrators can access
+
+## Single Feature
+**/api/authz/features/<:string>**
+
+Provide detailed information about a specific feature. Access is restricted to system administrators. The JSON response document is as follow
+
+```json
+{
+  "id": "withdrawItem",
+  "description": "The feature allows to withdrawn an item from the repository without deleting it. The restoreItem feature allow to undo the process",
+  "resourcetypes": [
+  	"item"
```

Comment:
I think having features locked to resource types might be limiting. I understand that we need to have somehow controlled values for them, but I may have a feature that doesn't relate with any of them, for instance "edit news" or "edit registry" are specific features for the ADMIN user and I think they doesn't apply for any of these resource types. Or perhaps I'm not seeing this properly.

Reply:
hi @paulo-graca ,
for such cases I expect that we would have a "administer" authorization or more fine-grains editNews, editRegistry that will apply to the "site" resourcetype

Reply:
Separately from this PR we need to hold a discussion of the wider application of access control and of finer-grained control.

Reply:
I'd recommend we start with something simple for now.  Tying a "feature" to a resource type seems OK for now -- as @abollini notes the `Site` resource type can be used for global features that apply at the site level.  

In the future (v8 or later) we may wish to revisit this (as @mwoodiupui notes) as we may want to eventually adopt an enhanced authorization / access control system.  However, that is out of scope for v7.